src/engine/engine.rs

```diff
@@ -26,8 +27,26 @@ pub trait Filesystem: HasName + HasUuid {
 }
 
 pub trait BlockDev: HasUuid {
-    /// path of the device node
+    /// Get the path of the device node for this device.
     fn devnode(&self) -> PathBuf;
+
+    /// Get the user-settable string associated with this blockdev.
+    fn user_id(&self) -> &Option<String>;
+
+    /// Set the user-settable string associated with this blockdev.
+    fn set_user_id(&mut self, user_id: Option<&str>) -> EngineResult<()>;
+
+    /// Get the hardware ID for this blockdev.
+    fn hardware_id(&self) -> &Option<String>;
```

Comment:
Should there be multiple hardware ids, should this have type ```Vec<String>```, or something richer?

Reply:
trying to initially keep it simple. I'm not sure how far we want to go for this feature, given it is not a critical one.